### PR TITLE
Feature: upgrade firmware option on NFC scan of df3 tag

### PR DESCRIPTION
**Pull request by rinat-enikeev**
Implemenbts RuuviFirmware Module and uses it in RuuviDiscover to enable firmware upgrade from add a sensor page

### CODE REVIEW

File: Modules/RuuviDiscover/Sources/RuuviDiscover/Resources/en.lproj/RuuviDiscover.strings [hunk 1]
```diff
@@ -28,3 +28,4 @@
 "add_sensor_nfc_df3_error" = "This tag cannot be added with NFC due to old firmware. Please add the tag with Bluetooth and update firmware.";
 "add_sensor_description" = "This page shows nearby Ruuvi sensors not yet added to the app. Tap a sensor to add it.";
 "add_sensor_via_nfc" = "Alternatively, you can add a sensor using NFC by selecting Add with NFC and touching it with your phone.";
+"upgrade_firmware" = "Upgrade Firmware";
```
Comment (priyonto) [inline Modules/RuuviDiscover/Sources/RuuviDiscover/Resources/en.lproj/RuuviDiscover.strings:31]: What about the translations for other languages?

Comment (rinat-enikeev) [reply Modules/RuuviDiscover/Sources/RuuviDiscover/Resources/en.lproj/RuuviDiscover.strings]: added https://github.com/ruuvi/com.ruuvi.station.ios/pull/1746/commits/a0b54918552567fb628b732b149c7d236d273963

File: Modules/RuuviFirmware/Sources/RuuviFirmware/Common/Spinner.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    public func updateUIView(_ uiView: UIActivityIndicatorView, context: Context) {
+        if isAnimating {
+            uiView.startAnimating()
+        } else {
+            uiView.stopAnimating()
+        }
+    }
+}
```
Comment (priyonto) [inline Modules/RuuviFirmware/Sources/RuuviFirmware/Common/Spinner.swift:26]: We have decided to use the new activity indicator across the app. So, we would probably need to replace this when we have the new activity indicator on the branch.

File: Modules/RuuviFirmware/Sources/RuuviFirmware/Util/RuuviColor.swift [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+//    static let ruuviTintColor = UIColor(named: "RuuviTintColor")
+//    static let ruuviTextColor = UIColor(named: "RuuviTextColor")
+//    static let ruuviMenuTextColor = UIColor(named: "RuuviMenuTextColor")
+//    static let ruuviLineColor = UIColor(named: "RuuviLineColor")
+//    static let ruuviSwitchDisabledTint =
+//        UIColor(named: "RuuviSwitchDisabledTint")
+//    static let ruuviSwitchEnabledTint =
+//        UIColor(named: "RuuviSwitchEnabledTint")
+//    static let ruuviSwitchDisabledThumbTint = UIColor(named: "RuuviSwitchDisabledThumbTint")
+}
```
Comment (priyonto) [inline Modules/RuuviFirmware/Sources/RuuviFirmware/Util/RuuviColor.swift]: Please clean this up before merging.

**Comment by rinat-enikeev** [comment]
<video src="https://github.com/ruuvi/com.ruuvi.station.ios/assets/722716/6c9e151b-7733-42b2-9b09-53eb5880456a" width=480/>

**Review by priyonto — APPROVED** [review]

File: Modules/RuuviFirmware/Sources/RuuviFirmware/Common/Spinner.swift (hunk 2 above)
Comment (priyonto) [reply Modules/RuuviFirmware/Sources/RuuviFirmware/Common/Spinner.swift:26]: Is this one used for spinner within the button?

Comment (rinat-enikeev) [reply Modules/RuuviFirmware/Sources/RuuviFirmware/Common/Spinner.swift:26]: Screenshot 2023-12-03 at 19 13 51

Comment (priyonto) [reply Modules/RuuviFirmware/Sources/RuuviFirmware/Common/Spinner.swift:26]: Okay, probably this can be replaced. But, no biggie. Thanks.

**Comment by rinat-enikeev** [comment]
fixes #1748

**Pull request merged**